sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -1388,3 +1389,66 @@ def test_unknown_category_that_are_negative():
     X_test_nan = np.asarray([[1, np.nan], [3, np.nan]])
 
     assert_allclose(hist.predict(X_test_neg), hist.predict(X_test_nan))
+
+
+@pytest.mark.parametrize(
+    "Hist", [HistGradientBoostingClassifier, HistGradientBoostingRegressor]
+)
+def test_pandas_categorical_results_same_as_ndarray(Hist):
+    """Check that pandas categorical give the same results as ndarray"""
+    pd = pytest.importorskip("pandas")
+
+    rng = np.random.RandomState(42)
+    n_samples = 5_000
+    n_cardinality = 50
+    max_bins = 100
+    f_num = rng.rand(n_samples)
+    f_cat = rng.randint(n_cardinality, size=n_samples)
+
+    # Make f_cat an informative feature
+    y = f_cat % 3 == 0
+
+    X = np.c_[f_cat, f_num]
```

Comment:
The ordering of the column is not neutral in case of correlated features (see: #26428 for more details on the kind of problems it can cause in practice).

So even if the columns values are independent in this test, I think we should avoid introducing any hard to debug problem by using the same column ordering here.

```suggestion
    X = np.c_[f_num, f_cat]
```

Reply:
Also assertion `assert len(predictor_1[0].nodes) == len(predictor_2[0].nodes)` would probably fail if we do not update this line while making `y` also depend on `f_num`.